setup.py

```diff
@@ -40,56 +47,89 @@ def run(self):
-        ('etc/jupyter/nbconfig/notebook.d', ['etc/jupyter/nbconfig/notebook.d/voila-gridstack.json']),
-        ('share/jupyter/nbextensions/voila-gridstack', ['voila-gridstack/static/extension.js',
-                                                        'voila-gridstack/static/voila-gridstack.js',
-                                                        'voila-gridstack/static/voila-gridstack.css',
-                                                        'voila-gridstack/static/gridstack.js',
-                                                        ])
+# Representative files that should exist after a successful build
+jstargets = [
+    os.path.join(lab_extension_source, "lib", "index.js"),
+    os.path.join(lab_extension_dest, "package.json"),
+]
+
+package_data_spec = {
+    name: [
+        "*"
     ]
-    # Add all the templates
-    for root, dirs, files in os.walk('share'):
-        root_files = [os.path.join(root, i) for i in files]
-        data_files.append((root, root_files))
-
-    return data_files
-
-
-setup_args = {
-    'name': 'voila-gridstack',
-    'version': '0.0.12',
-    'packages': find_packages(),
-    'data_files': get_data_files(),
-    'package_data': {
-        'voila-gridstack': [
-            'static/*'
-        ]
-    },
-    'install_requires': [
-        'voila>=0.2.0,<0.3.0',
+}
+
+data_files_spec = [
+    ("share/jupyter/labextensions/%s" % labext_name, lab_extension_dest, "**"),
+    ("share/jupyter/labextensions/%s" % labext_name, HERE, "install.json"),
+    ("etc/jupyter/jupyter_server_config.d", "etc/jupyter/jupyter_server_config.d", "voila-gridstack.json"),
+    ("etc/jupyter/jupyter_notebook_config.d", "etc/jupyter/jupyter_notebook_config.d", "voila-gridstack.json"),
+    ("etc/jupyter/nbconfig/notebook.d", "etc/jupyter/nbconfig/notebook.d", "voila-gridstack.json"),
+    ("share/jupyter/nbextensions/voila-gridstack", "voila-gridstack/static", "**")
+]
+
+cmdclass = create_cmdclass("jsdeps",
+    package_data_spec=package_data_spec,
+    data_files_spec=data_files_spec
+)
+
+cmdclass["jsdeps"] = combine_commands(
+    install_npm(lab_extension_source, build_cmd="build:prod", npm=["jlpm"]),
+    ensure_targets(jstargets),
+    DevelopCmd
+)
+
+with open("README.md", "r") as fh:
+    long_description = fh.read()
+
+setup_args = dict(
+    name=name,
+    version="0.0.12",
+    url="https://github.com/voila-dashboards/voila-gridstack",
+    author="Voila Development Team",
+    author_email="jupyter@googlegroups.com",
+    description="A GridStack template for Voila.",
+    long_description= long_description,
+    long_description_content_type="text/markdown",
+    cmdclass=cmdclass,
+    packages=setuptools.find_packages(),
+    install_requires=[
+        "jupyterlab_widgets>=1.0.0a6",
+        "voila>=0.2.0,<0.3.0"
     ],
-    'extras_require': {
-        'test': [
+    extras_require={
+        "test": [
             'ipykernel',
+            'jupyter_server~=1.0.1',
```

Comment:
Sticking to this for now since a recent version of `jupyter_server` changed the name of the pytest fixtures.